src/workspaces/features/WorkspaceConnectionPicker/WorkspaceConnectionPicker.tsx

```diff
@@ -74,20 +74,24 @@ const WorkspaceConnectionPicker = (props: WorkspaceConnectionPickerProps) => {
     [],
   );
 
+  const comboBoxValue = useMemo(() => {
+    return options.find((option) => option.slug === value) ?? null;
+  }, [value, options]);
+
   return (
     <Combobox
       required={required}
       onChange={onChange}
       loading={loading}
       withPortal={withPortal}
       displayValue={displayValue}
-      by={(a: Option, b: Option) => a.id === b.id}
+      by="name"
```

Comment:
Looks weird to me that we use the "name" of the connection as the identifier of the connection.